sda/cmd/api/api.go

```diff
@@ -389,3 +392,53 @@ func listUserFiles(c *gin.Context) {
 	c.Writer.Header().Set("Content-Type", "application/json")
 	c.JSON(200, files)
 }
+
+// addHashedKey handles the addition of a hashed key to the database.
+// It expects a JSON payload containing the key hash and its description.
+// If the JSON payload is invalid, it responds with a 400 Bad Request status.
+// If there is no row update in the database, it responds with a 409 Conflict status
+// If the database insertion fails, it responds with a 500 Internal Server Error status.
+// On success, it responds with a 200 OK status.
+func addHashedKey(c *gin.Context) {
+	var keyhash schema.KeyhashInsertion
+	if err := c.BindJSON(&keyhash); err != nil {
+		c.AbortWithStatusJSON(
+			http.StatusBadRequest,
+			gin.H{
+				"error":  "json decoding : " + err.Error(),
+				"status": http.StatusBadRequest,
+			},
+		)
+
+		log.Errorf("Invalid JSON payload: %v", err)
+
+		return
+	}
+
```

Comment:
At this point we should do validation of the payload data i.e ensure that it is a c4gh public key before we hash it and store it in the database.